guidelines: add leaving organization check-list

0xc0170 opened the pull request:
Any person leaving our organization should follow these guidelines.

@AnotherButler Please review - I added new section to adding yourself to the organization - leaving it.

0xc0170 committed:
guidelines: add leaving organization check-list

Any person leaving our organization should follow these guidelines.

0xc0170 commented:
@ARMmbed/mbed-os-maintainers

adbridge commented on docs/reference/contributing/guidelines/guidelines.md:
```diff
@@ -46,6 +46,10 @@ If so, you can request to become an organization member, but you must first ensu
+### Leaving ARMmbed organization
+
+Best practices for leaving the organization are documented on [Github help](https://help.github.com/en/articles/best-practices-for-leaving-your-company). Please follow these best practices once you leave the organization.
+
```
I would rewrite this to say something like:

When leaving the company please follow the following best practice: <link>

0xc0170 replied:
fixed

Commit:
Edit guidelines.md

Copy edit file for inclusion of articles, correct comma use and removal of repetition.

AnotherButler reviewed (approved):
Thanks for the PR 👍

0xc0170 commented:
Ready!
